homeassistant/helpers/template.py

```diff
@@ -2093,6 +2152,10 @@ def wrapper(_: Any, *args: _P.args, **kwargs: _P.kwargs) -> _R:
 
             return pass_context(wrapper)
 
+        self.globals["internal_url"] = hassfunction(_get_internal_url)
+        self.globals["external_url"] = hassfunction(_get_external_url)
+        self.filters["sign_path"] = hassfunction(_sign_path)
```

Comment:
I don't agree with exposing this to the template engine, as it might expose a security risk.

Reply:
There is currently no other way to generate signed URLs for notifications. The `notify` platform has been a state of flux for a while so there is no real solution in there that can automatically sign URLs.

Do you have a different suggestion/solution for generating signed URLs for notifications? I could expose [async_process_play_media_url](https://github.com/home-assistant/core/blob/dev/homeassistant/components/media_player/browse_media.py#L29) instead of internal/external URL, but that is essentially do the same thing adding all 3 of these.

Reply:
>  The notify platform has been a state of flux for a while so there is no real solution in there that can automatically sign URLs.

Let's improve that instead of working around things. 🤷 

> Do you have a different suggestion/solution for generating signed URLs for notifications?

I don't, but this does impose a security risk, as template (Blueprint) can open up/sign any URL at will. Which IMHO is not something that belongs here.